plugins/ua_pubsub_udp.c

```diff
@@ -177,6 +178,20 @@ UA_PubSubChannelUDPMC_open(const UA_PubSubConnectionConfig *connectionConfig) {
     memcpy(channelDataUDPMC->ai_addr, rp->ai_addr, sizeof(*rp->ai_addr));
     newChannel->handle = channelDataUDPMC; /* Link channel and internal channel data */
 
+    if(address.networkInterface.length > 0) {
+        channelDataUDPMC->intf_addr = (struct sockaddr_storage *)
+            UA_calloc(1, sizeof(struct sockaddr_storage));
+        if(!channelDataUDPMC->intf_addr) {
+            UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER,
+                        "PubSub Connection creation failed. Out of memory.");
+            UA_close(newChannel->sockfd);
```

Comment:
Clean up `channelDataUDPMC->ai_addr`  in this error case.
In the error case below clean up both `channelDataUDPMC->ai_addr` and  `channelDataUDPMC->intf_addr`.